src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy

```diff
@@ -239,4 +213,88 @@ class Oauth2IT extends AbstractIT {
             .body("message", not(isEmptyOrNullString()))
             .body("error", not(isEmptyOrNullString()))
     }
+
+    /** We should be able to use the client_credentials grant type to get an access token
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/8">#8</a>
+     */
+    @Test(groups=["v100", "json"])
+    public void oauthClientCredentialsGrantSucceeds() throws Exception {
+        TestAccount testAccount = new TestAccount()
+        testAccount.registerOnServer()
+        deleteOnClassTeardown(testAccount.href)
+
+        // Get API keys so we can use it for client credentials
+
+        Response apiKeysResource = given()
+            .header("User-Agent", "stormpath-framework-tck")
+            .header("Authorization", RestUtils.getBasicAuthorizationHeaderValue())
+            .port(443)
+        .when()
+            .post(testAccount.href + "/apiKeys")
+        .then()
+            .statusCode(201)
+        .extract()
+            .response()
+
+        String apiKeyId = apiKeysResource.body().jsonPath().getString("id")
+        String apiKeySecret = apiKeysResource.body().jsonPath().getString("secret")
+
+        // Attempt to get tokens
+
+        given()
+            .param("grant_type", "client_credentials")
+            .auth()
+                .preemptive().basic(apiKeyId, apiKeySecret)
+            .contentType(ContentType.URLENC)
+        .when()
+            .post(OauthRoute)
+        .then()
+            .statusCode(200)
+            .contentType(ContentType.JSON)
```

Comment:
That's why I wrote it, but actually `client_credentials` should not be outputting a refresh token. I can write a separate spec but it's not really necessary...